### PR TITLE
secondary filtering

### PR DESCRIPTION
**Pull request by drahnr**
Adds some extra checks.

TODO:

* [x] test case

### CODE REVIEW

File: runtime/parachains/src/disputes.rs [hunk 1]
```diff
@@ -1001,6 +1027,10 @@ impl<T: Config> Pallet<T> {
 						// by the duplicate checks above. It's only the first one which
 						// may not already have been filtered out.
 						filter.remove_index(first_index_in_set);
+
+						// Removing individual statments can cause the dispute to become onesided.
+						// Checking that (again) is done after the loop. Remove the bit indices.
+						summary.new_flags.set(validator_index.0 as _, false);
```
Comment (eskimor) [inline runtime/parachains/src/disputes.rs:1033]: Why is this suddenly necessary?

File: runtime/parachains/src/disputes.rs [hunk 2]
```diff
@@ -1026,6 +1056,37 @@ impl<T: Config> Pallet<T> {
...
+						// Note that this does not distinguish between pro or cons votes,
+						// since allowing both of them, even if the spam threshold would be reached
+						// is a good thing.
+						// Overflow is no concern, disputes are limited by weight.
+						DisputeStatement::Valid(_) => {
+							if Some(true) == summary.new_flags.get(v_i) {
+							vote_for_count += 1;
+						}}
+						DisputeStatement::Invalid(_) => {
+							if Some(true) == summary.new_flags.get(v_i) {
```
Comment (drahnr) [inline runtime/parachains/src/disputes.rs:1080]: @eskimor it's used here (and nowhere else) to make sure we only count newly added votes, which wasn't done before

File: runtime/parachains/src/disputes.rs (hunk 1 above)
Comment (drahnr) [reply runtime/parachains/src/disputes.rs:1033]: See my next comment, it's not used outside the scope of this `fn` so it's alright to modify it at this point.

File: runtime/parachains/src/disputes.rs [hunk 3]
```diff
@@ -652,8 +670,12 @@ impl<BlockNumber: Clone> DisputeStateImporter<BlockNumber> {
 					.map(|i| (ValidatorIndex(i as _), SpamSlotChange::Dec))
 					.collect()
 			},
-			(true, true) | (true, false) => {
-				// nothing to do. (true, false) is also impossible.
+			(true, false) => {
+				log::error!("Dispute statments are never removed. qed");
```
[the remark this answers is not in the document]
Comment (drahnr) [reply runtime/parachains/src/disputes.rs:674]: It's both, but it's also the runtime so we don't want to panic if some unrelated code change _might_ make this possible in the future.

File: runtime/parachains/src/disputes.rs (hunk 1 above)
Comment (eskimor) [reply runtime/parachains/src/disputes.rs:1036]: Now, that I check this more closely, it makes even less sense. `new_flags` has type: [DisputeStateFlags](https://github.com/paritytech/polkadot/blob/31a254b9174388422f9fe94a831c90a64f4c172c/runtime/parachains/src/disputes.rs#L498), I don't understand why we would use `ValidatorIndex` as index here. :confused:

Comment (drahnr) [reply runtime/parachains/src/disputes.rs:1036]: Yeah, you're correct. That makes no sense at all.

**Comment by dvdplm** [comment]
A better title&description would be good here.

**Review by tdimitrov — APPROVED** [review]
I had a code walkthrough with @drahnr. The change is reasonable, tests are passing => should be good to go.

**Review by eskimor — APPROVED** [review]

**Pull request merged**